gateway/gateway.go

```diff
@@ -284,13 +356,37 @@ func (gateway *HandleT) webRequestBatchDBWriter(process int) {
 	}
 }
 
-func (gateway *HandleT) dedupWithBadger(body *[]byte, messageIDs [][]byte, writeKey string, writeKeyDupStats map[string]int) {
-	var toRemoveMessageIndexes []int
+func appendToSet(set map[string]struct{}, elements []string) {
+	for _, element := range elements {
+		set[element] = struct{}{}
+	}
+}
+
+func (gateway *HandleT) dedup(body *[]byte, messageIDs []string, allMessageIDsSet map[string]struct{}, writeKey string, writeKeyDupStats map[string]int) {
+	toRemoveMessageIndexesSet := make(map[int]struct{})
+	//Dedup within events batch in a web request
+	for idx, messageID := range messageIDs {
```

Comment:
Should we change this too to set? Just to avoid O(n2) (won't matter much though the max batch size is 32 anyway)

```
	for idx, messageID := range messageIDs {
     messageIDSet[messageID] = struct{}{}
  }

	for idx, messageID := range messageIDs {
      		if _, ok := messageIDSet[messageID]; ok {
			toRemoveMessageIndexesSet[idx] = struct{}{}
		}
}


```

Reply:
There is a problem with the logic above. It will remove all the repeated messageIDs.
For example, consider this batch. [m1, m2, m1, m3, m4]
messageIDSet becomes [m1, m2, m3, m4]
toRemoveMessageIndexesSet will become [0, 1, 2, 3, 4] and all messages will be removed.

Reply:
Thinking more, we can remove this "Dedup within events batch in a web request" logic altogether. Because this batch [m1, m2, m1, m3, m4] scenario will never arise.